ctapipe/image/cleaning.py

```diff
@@ -67,16 +73,123 @@ def tailcuts_clean(geom, image, picture_thresh=7, boundary_thresh=5,
     # matrix (2d), we find all pixels that are above the boundary threshold
     # AND have any neighbor that is in the picture
     pixels_above_boundary = image >= boundary_thresh
-    pixels_with_picture_neighbors = geom.neighbor_matrix_sparse.dot(
-        pixels_in_picture)
+    pixels_with_picture_neighbors = geom.neighbor_matrix_sparse.dot(pixels_in_picture)
     if keep_isolated_pixels:
-        return (pixels_above_boundary
-                & pixels_with_picture_neighbors) | pixels_in_picture
+        return (
+            pixels_above_boundary & pixels_with_picture_neighbors
+        ) | pixels_in_picture
     else:
         pixels_with_boundary_neighbors = geom.neighbor_matrix_sparse.dot(
-            pixels_above_boundary)
-        return ((pixels_above_boundary & pixels_with_picture_neighbors) |
-                (pixels_in_picture & pixels_with_boundary_neighbors))
+            pixels_above_boundary
+        )
+        return (pixels_above_boundary & pixels_with_picture_neighbors) | (
+            pixels_in_picture & pixels_with_boundary_neighbors
+        )
+
+
+def mars_image_cleaning(
+    geom,
+    image,
+    picture_thresh=7,
+    boundary_thresh=[5, 5],
+    keep_isolated_pixels=False,
+    min_number_picture_neighbors=0,
+):
+    """
+    Clean an image by selection pixels that pass a three-threshold tail-cuts
+    procedure.
+    All thresholds are defined with respect to the pedestal
+    dispersion. All pixels that have a signal higher than the core (picture)
+    threshold will be retained, along with all those above the boundary
+    threshold that are neighbors of a core pixel AND all those above
+    the boundary threshold that are neighbors of a neighbor of a core pixel.
+
+    Parameters
+    ----------
+    geom: `ctapipe.instrument.CameraGeometry`
+        Camera geometry information
+    image: array
+        pixel values
+    picture_thresh: float or array
+        threshold above which all pixels are retained
+    boundary_thresh: array
+        array of 2 thresholds, first the one above which pixels are retained if
+        they have a neighbor already above the picture_thresh, then the one
+        applied iteratively to the neighbor of the neighbor
+    keep_isolated_pixels: bool
+        If True, pixels above the picture threshold will be included always,
+        if not they are only included if a neighbor is in the picture or
+        boundary
+    min_number_picture_neighbors: int
+        A picture pixel survives cleaning only if it has at least this number
+        of picture neighbors. This has no effect in case keep_isolated_pixels is True
+
+    Returns
+    -------
+    A boolean mask of *clean* pixels.  To get a zero-suppressed image and pixel
+    list, use `image[mask], geom.pix_id[mask]`, or to keep the same
+    image size and just set unclean pixels to 0 or similar, use
+    `image[~mask] = 0`
```

Comment:
Imo `mask` does not very well express whether it contains the pixels to be discarded, or the pixels surviving the cleaning. The example is necessary to understand what is returned. I think `survivors`or `signal` might be better. This might be a comment for all cleaning functions.